### PR TITLE
Update repr html links for events.

### LINKED ISSUE #1552: Update event links in repr_html
The repr_html links for demographic events currently point to the class documentation. However, we're not documenting the classes any more (except for legacy), so some of these links will now be broken. (See #1550)

Change the links to that they point to the appropriate Demography.add_x method, e.g., change from pointing to to class documentation for MassMigration to method documentation for ``Demography.add_mass_migration``.

### PR DESCRIPTION
**Pull request by jeromekelleher**
Closes #1552

### CODE REVIEW

File: msprime/demography.py [hunk 1]
```diff
@@ -744,10 +745,12 @@ def _events_html(self, events, title=None):
         data = []
         for event in events:
             class_name = event.__class__.__name__
+            camel_case = re.sub(r"(?<!^)(?=[A-Z])", "_", class_name).lower()
+            add_method = f"msprime.Demography.add_{camel_case}"
             # TODO change this to stable when 1.0 is released.
             type_html = (
-                "<a href='https://tskit.dev/msprime/docs/latest/api.html#msprime."
-                f"{class_name}'>{event._type_str}</a>"
+                "<a href='https://tskit.dev/msprime/docs/latest/api.html#"
```
Comment (benjeffery) [inline msprime/demography.py:752]: Should this normally be stable? Do we change this after the pre-releases are over?

**Review by benjeffery — APPROVED** [review]
LGTM, one comment

File: msprime/demography.py (hunk 1 above)
Comment (jeromekelleher) [reply msprime/demography.py:752]: Yep, agreed. That's what the TODO just above is supposed to say :wink:

Comment (benjeffery) [reply msprime/demography.py:752]: Haha, bad idea to review on mobile!!

**Pull request merged**